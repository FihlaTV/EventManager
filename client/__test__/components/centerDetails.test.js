/* global expect afterEach beforeEach describe it */
import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {
  CenterDetailsComponent,
  mapDispatchToProps,
  mapStateToProps,
} from '../../src/components/centerDetailsComponent';

configure({ adapter: new Adapter() });

describe('Tests for centerDetails component', () => {
  const props = {
    match: {
      params: {
        id: 1,
      },
    },
    setPage: jest.fn(),
    centerDetails: [],
    facilities: [],
    history: {},
    currentPage: 1,
  };

  it('should render self without crashing', () => {
    const wrapper = shallow(<CenterDetailsComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should call setPage function when comopnent unmounts', () => {
    const wrapper = shallow(<CenterDetailsComponent {...props} />);
    wrapper.unmount();
    expect(props.setPage).toHaveBeenCalled();
  });
});
describe('mapStateToProps', () => {
  const state = {
    centers: {
      centerDetails: {},
    },
    facilities: {
      facilities: {},
    },
    page: {
      currentPage: {},
    },
  };
  it('returns the expected properties', () => {
    const mstp = mapStateToProps(state);
    expect(mstp).toHaveProperty('centerDetails');
    expect(mstp).toHaveProperty('facilities');
    expect(mstp).toHaveProperty('currentPage');
  });
});

describe('mapDispatchToProps', () => {
  it('should return', () => {
    const dispatch = jest.fn();
    // const spy = jest.fn();
    const mdtp = mapDispatchToProps(dispatch);
    expect(mdtp).toHaveProperty('setCenterDetails');
    expect(mdtp).toHaveProperty('setPage');
    // expect(mdtp.setCenterDetails).toHaveBeenCalled();
  });
});
