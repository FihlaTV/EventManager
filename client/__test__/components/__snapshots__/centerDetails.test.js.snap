// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for centerDetails component should render self without crashing 1`] = `
<div>
  <h2>
    Loading...
  </h2>
</div>
`;
